### ISSUE TITLE
DotPlot split.by error

### ISSUE THREAD

**Issue by craigamartens** [satijalab/seurat#1634]
Hi,  I have 3 datasets that I integrated and now trying to display a dot plot by splitting by the 3 datasets.  

`dp <- DotPlot(subset3.integrated, features = c('Itgam', 'Il7r', 'Kit'), group.by = "predicted.id", split.by = "stim", cols = c("red", "blue", "green"))`

when I run this I get "not enough colors for the number of groups".

if I add one more color, I get "Error in FUN(X.... subscript out of bounds"

one odd thing is that when I do a TSNE and group.by my 3 datasets, I get the 3 back plus "NA".  Not sure if that is the root of the issue or not?

Thanks for any help.
Craig

**Comment by andrewwbutler** [comment]
Are you able to provide the object (or downsampled version) that reproduces the issue? Also, what happens if you remove the NA cells?

**Comment by craigamartens** [comment]
I might be able to supply the object.  So a little update.  I was able to remove the NA cells using subset, but that still didn't completely solve the problem.  I was able to get the split.by to work if I didn't use the group.by also.  I would like to display the dot plot based on the predicted cell types, using TransferData/AddMetaData, instead of the original clustering.

**Comment by craigamartens** [comment]
The first 2 run fine, the third doesn't.

`DotPlot(subset4.integrated, features = c('Itgam', 'Il7r', 'Kit', 'Flt3', 'Cd34', 'Slamf1'), split.by = "stim", cols = c("red", "blue", "green"))
DotPlot(subset4.integrated, features = c('Itgam', 'Il7r', 'Kit', 'Flt3', 'Cd34', 'Slamf1'), group.by = "predicted.id", cols = c("red", "blue"))
DotPlot(subset4.integrated, features = c('Itgam', 'Il7r', 'Kit', 'Flt3', 'Cd34', 'Slamf1'), group.by = "predicted.id", split.by = "stim", cols = c("red", "blue", "green"))
`

[subset.gz](https://github.com/satijalab/seurat/files/3282371/subset.gz)

**Comment by andrewwbutler** [comment]
This doesn't appear to the the right object. Can you double check that you uploaded the correct data? This one is a Seurat v2 object without those features or the stim or predicted.id meta.data columns.

**Comment by craigamartens** [comment]
Hi,  I just downloaded from the link above, renamed to subset.Robj and then ran the following:

```
> load("subset.Robj", verbose = TRUE)
Loading objects:
  subset4.integrated
> head(subset4.integrated@meta.data, 5)
                 orig.ident nCount_RNA nFeature_RNA stim
CAGAGAGAGAGGGCTT       INF2       9676          506 inf2
TGCCCTATCAGGTAAA       INF2       7924          405 inf2
CCTCAGTAGGAGTACC       MOCK       9197          519 mock
TACTTGTGTAGGCATG       MOCK       6523          211 mock
AGCTCCTCAATGGAGC       INF1       6487          515 inf1
                 integrated_snn_res.0.45 seurat_clusters    predicted.id
CAGAGAGAGAGGGCTT                       0               0 proerythroblast
TGCCCTATCAGGTAAA                       0               0 proerythroblast
CCTCAGTAGGAGTACC                       0               0 proerythroblast
TACTTGTGTAGGCATG                       0               0 proerythroblast
AGCTCCTCAATGGAGC                       0               0 proerythroblast


```

**Comment by mojaveazure** [comment]
This should be fixed in Seurat v3.0.2

**Comment by craigamartens** [comment]
Thanks.  That did the trick.
